### DISCUSSION TITLE
envoy filter namespace

### DISCUSSION THREAD

**Discussion by PriyankaAdama** [open-policy-agent/community#368] [Community]
Followed istio plugin turorial

Team, I am trying to understand about envoy filter here https://github.com/open-policy-agent/opa-envoy-plugin/blob/main/examples/istio/quick_start.yaml#L8 . Tried to  have the filter istio-system namespace and I see OPA is receiving the requests at all and always returns 200 response.

Whereas, if am enabling authz filter for the service namespace , as an example in spite of istio-system to foo it always gives me 403. Can you please help understand about this filter. I am trying to inject it into all my services and not into one gateway service

**Comment by ashutosh-narkar** [comment]
If you want to learn more about the EnvoyFilter resource itself check [this](https://istio.io/latest/docs/reference/config/networking/envoy-filter/) out. In terms of the OPA-Istio tutorial, if you think you are not getting the policy decisions you expect, it would be helpful if you shared the OPA policy and the input OPA gets from Envoy. More info on how that input looks can be found [here](https://www.openpolicyagent.org/docs/latest/envoy-primer/#input-document).  We would recommend you [unit test](https://www.openpolicyagent.org/docs/latest/policy-testing/) your policies before you apply them in the system.
